### PR TITLE
save form data to local storage

### PR DESCRIPTION
**Pull request by ericmahare**
In this section, I did the following: -
- Add user data to the local storage 
- fetch data from the local storage to the input fields when the page loads

### CODE REVIEW

File: assets/js/main.js [hunk 1]
```diff
@@ -91,6 +93,20 @@ const projects = [
...
+    userName.value = '';
+  } else {
+    const localData = localStorage.getItem('user');
+    const data = JSON.parse(localData);
+
+    // set localstorage data to the input values
+    email.value = data.email;
+    message.value = data.message;
+    userName.value = data.name;
+  }
```
Comment (ShadyShawkat) [inline assets/js/main.js:109]: - [ ] Kindly make sure to preserve the form data in local storage. You're currently checking if the local storage has data and populating the fields with that data. However, you're not saving the data or setting anything in the local storage if there's nothing already there. 
Please make sure to add this functionality. (You can use the `onChange` event to get data from the forum and save them in the local storage with every user input)

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: assets/js/main.js [hunk 2]
```diff
@@ -254,6 +270,17 @@ const similarTask = () => {
...
+const storeToLocalStorage = () => {
+  const data = {
+    name: userName.value,
+    email: email.value,
+    message: message.value,
+  };
+  const storageData = JSON.stringify(data);
+  localStorage.setItem('user', storageData);
+};
+
```
Comment (ericmahare) [inline assets/js/main.js]: Hello, thanks for your review. Above is the function to save the user data when the form is being submitted.  In the section that you have included, I was just making sure that the local storage data is available before fetching it.

Comment (ShadyShawkat) [reply assets/js/main.js]: Hey @ericmahare,
The requirement says to save the data when an input value is changed by the user, not when submitting the form.
image

Comment (ericmahare) [reply assets/js/main.js]: Thanks, I will rectify that mistake

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**